engines/bastion_katello/test/organizations/organization-selector.controller.test.js

```diff
@@ -16,8 +16,11 @@ describe('Controller: OrganizationSelectorController', function() {
         $state = $injector.get('$state');
         $window = {location: {href: ''}};
 
+        var selectFunction = function (func) { func.call() };
         Organization.select = function (params) {
-            return {'$promise': {catch: function (func) { func.call() } }};
+            return {'$promise': { then:selectFunction,
+                                 catch: selectFunction }
```

Comment:
I'm assuming this is something similar to [$q](https://docs.angularjs.org/api/ng/service/$q) ?  Doesn't make much sense to me, but I'll trust you since you're the Angular expert ;)

Looks like `catch` isn't being used anymore?  (even if it is, doesn't make sense to me to have the same callback for `then` and `catch`)

Reply:
I m trying to mock the promise method in the unit test and have it run accordingly. I dont think the catch is necessary here going to remove.

Reply:
updated